### PR TITLE
Page builder Compatibility

### LINKED ISSUE #5: Page Builder Compatibility
Would you be open to allowing page builder compatibility? We'd have to do something similar to this: https://github.com/ronalfy/twentynineteen-child/tree/beaverpartsupport

We'd split the header and footer, and add hooks for both, and allow page builders like Beaver Builder and Elementor to customize the theme's header, footer, etc.

I can put together a PR concept for this if you're open to this.

### PR DESCRIPTION
**Pull request by ronalfy**
Resolves #5 

screencapture-localhost-wordpress-2019-07-24-15_40_11

The following changes make the theme page builder friendly.

The header/footer are modular and are included in functions.php.

I've created a sample child theme that adds Beaver Builder compatibility, for example.

https://github.com/ronalfy/jarvis-beaver-builder-child

The attached screenshot shows a custom header/footer for the site using Beaver Builder, but can be extended to other page builders as well.

Will need a fully stretched page template that will accommodate adding content in. Basically, a blank canvas that includes the header and footer and a full-width content area in between.

### CODE REVIEW

File: header.php [hunk 1]
```diff
@@ -23,7 +23,13 @@
 
 <body <?php body_class(); ?>>
 
-<?php wp_body_open(); ?>
+<?php
+if ( function_exists( 'wp_body_open' ) ) {
```
Comment (BinaryMoon) [inline header.php]: Could we remove this commit please? I've added compatability for wp_body_open here:
https://github.com/BinaryMoon/jarvis/blob/master/inc/compat.php

File: style.css [hunk 2]
```diff
@@ -3227,15 +3227,32 @@ a.header-image img {
 	.menu-primary > .menu-wrap > li > a {
 		font-weight: bold;
 		line-height: 2.5;
+<<<<<<< HEAD
```
Comment (BinaryMoon) [inline style.css]: I don't know what these are called but there seems to be a bunch of merge conflict messages committed here :)

Running npm run build will rebuild the css and remove this.

File: functions.php [hunk 3]
```diff
@@ -107,3 +107,17 @@
 if ( class_exists( 'WooCommerce' ) ) {
 	require get_parent_theme_file_path( 'inc/woocommerce.php' );
 }
+add_action( 'jarvis_header', 'jarvis_include_header' );
+/**
+ * Include the default site header for the site.
+ */
+function jarvis_include_header() {
+	include 'inc/header.php';
```
Comment (BinaryMoon) [inline functions.php]: could we change this to `get_template_part( 'parts/header.php' )`?

Using get_template_part means the files can be filtered/ overridden by child themes.

In addition template parts go in the parts directory, and code (filters and reusable functions) go in inc. I should add something to the readme.md about the folder structure.

File: header.php [hunk 4]
```diff
@@ -23,53 +23,12 @@
...
-
-	</header>
-
-	<div class="container" id="site-content">
+if ( function_exists( 'wp_body_open' ) ) {
+	wp_body_open();
+} else {
+	do_action( 'wp_body_open' );
+}
+do_action( 'jarvis_before_header' );
```
Comment (BinaryMoon) [inline header.php]: This is one of the things I don't understand about adding page build compat. Why are there so many filters? Can't the same thing be accomplished with one?

**Comment by BinaryMoon** [comment]
Thanks for this Ron! I find the screenshot amazing really cool. I've never used a page builder and didn't realise you could do things quite like this. :)

I've added some comments to your commits.

Thanks!

File: header.php (hunk 1 above)
Comment (ronalfy) [reply header.php]: I've made the requested change with the updated PR.

File: functions.php (hunk 3 above)
Comment (ronalfy) [reply functions.php]: I have moved these to template parts.

File: header.php (hunk 4 above)
Comment (ronalfy) [reply header.php]: It's so you can add items before the header (like a top nav) and then display the regular header. For example, I can create a header template, and then add a template part for breadcrumbs that will show right after the header.

File: style.css (hunk 2 above)
Comment (ronalfy) [reply style.css]: I'm not sure why there are merge conflicts. I updated the PR with your latest changes in Master.

**Comment by ronalfy** [comment]
Here's an updated screenshot showing breadcrumbs. I've also attached a video of how a page builder might be used. I've also included a template that is content only so a page builder can take over the content.

screencapture-localhost-wordpress-blog-event-tracking-test-2019-07-25-17_29_06

https://www.dropbox.com/s/1rjimbh4kswmaun/jarvis-page-builder.mp4?dl=0

**Comment by ronalfy** [comment]
Something that GeneratePress does is it adds a metabox to the side of the post/page that allows you to have a stretched layout and will disable the title. If this is approved, I can work on adding that functionality in.

**Comment by BinaryMoon** [comment]
All done! :)

**Pull request merged**